src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -128,11 +129,18 @@ public override void OnStopActivity(Activity activity, object payload)
                     return;
                 }
 
-                var response = context.Response;
-                activity.AddTag(SemanticConventions.AttributeHttpStatusCode, response.StatusCode.ToString());
+                if (this.TryGetGrpcMethod(activity, out var grpcMethod))
+                {
+                    this.AddGrpcAttributes(activity, grpcMethod, context);
+                }
+                else
+                {
+                    var response = context.Response;
+                    activity.AddTag(SemanticConventions.AttributeHttpStatusCode, response.StatusCode.ToString());
 
-                Status status = SpanHelper.ResolveSpanStatusForHttpStatusCode(response.StatusCode);
-                activity.SetStatus(status.WithDescription(response.HttpContext.Features.Get<IHttpResponseFeature>()?.ReasonPhrase));
+                    Status status = SpanHelper.ResolveSpanStatusForHttpStatusCode(response.StatusCode);
+                    activity.SetStatus(status.WithDescription(response.HttpContext.Features.Get<IHttpResponseFeature>()?.ReasonPhrase));

```

Comment:
It does support null. Using WithDescription results in a new Status being instantiated which would be gratuitous in the event the description is null. This code is not new to this PR, though I can put in a check here just to be more explicit.

Reply:
I'd actually prefer to put the check in a separate PR unless you feel strongly otherwise. It's not a big change, but should probably include an effort to review any tests surrounding the change.